pkg/async/events/implementations/node_execution_event_writer.go

```diff
@@ -0,0 +1,41 @@
+package implementations
+
+import (
+	"context"
+
+	"github.com/flyteorg/flyteadmin/pkg/async/events/interfaces"
+	"github.com/flyteorg/flyteadmin/pkg/repositories"
+	"github.com/flyteorg/flyteadmin/pkg/repositories/transformers"
+	"github.com/flyteorg/flyteidl/gen/pb-go/flyteidl/admin"
+	"github.com/flyteorg/flytestdlib/logger"
+)
+
+type nodeExecutionEventWriter struct {
+	db     repositories.RepositoryInterface
+	events chan admin.NodeExecutionEventRequest
+}
+
+func (w *nodeExecutionEventWriter) Write(event admin.NodeExecutionEventRequest) {
+	w.events <- event
+}
+
+func (w *nodeExecutionEventWriter) Run() {
+	for event := range w.events {
+		eventModel, err := transformers.CreateNodeExecutionEventModel(event)
+		if err != nil {
+			logger.Warnf(context.TODO(), "Failed to transform event [%+v] to database model with err [%+v]", event, err)
+			continue
+		}
+		err = w.db.NodeExecutionEventRepo().Create(context.TODO(), *eventModel)
```

Comment:
can we add a comment here explaining that we will drop the event on the floor and why it is ok to be lossy!

Reply:
done